parsons/ngpvan/van.py

```diff
@@ -54,19 +55,18 @@ class VAN(
     `Args:`
         api_key : str
             A valid api key Not required if ``VAN_API_KEY`` env variable set.
-        auth_name: str
-            Should not pass this argument
         db: str
             One of ``MyVoters``, ``MyMembers``, ``MyCampaign``, or ``EveryAction``
-        uri: str
-            Base uri to make api calls.
-        raise_for_status: boolean
-            Raise excection when encountering a 4XX or 500 error.
     `Returns:`
         VAN object
     """
 
-    def __init__(self, api_key=None, auth_name="default", db=None, raise_for_status=True):
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        db: Optional[Literal["MyVoters", "MyCampaign", "MyMembers", "EveryAction"]] = None,
+        **_,  # ignore any other keywords (previously this signature included other keywords)
```

Comment:
this is backwards compatible in that if people have code where they did set `auth_name` or `raise_for_status`, it won't throw an exception.

it could be better:
1. remove it entirely, and let the code throw an exception since if people were passing those arguments it wasn't doing anything, and they might want to be alerted to do that. it would also keep this signature clean.
2. not have a catchall here, but explicitly check for `auth_name` and `raise_for_status` and raise a warning if they are set. this would make the signature kind of messy, but it would avoid the catchall.

Reply:
Personally I think I'd rather see that argument removed, I think it's more confusing and kind of a footgun to have an unused kwargs. I think losing backwards compatibility with arguments that were already fake is kind of OK

Reply:
But yeah probably "best practice" would be to leave the arguments in place and issue a deprecationwarning if they're included, and then remove them in the next major release. 🤷🏻

Reply:
but yeah I vote option (1)

Reply:
Do we know what the previous use of `auth_name` or `raise_for_status` was? What's the likelihood we'll catch people out on this?

@sharinetmc @KasiaHinkson any way you can take a quick look and see if this would cause issues within TMC?

Reply:
they were unused in the function body.

* `raise_for_status` hasn't done anything six years https://github.com/move-coop/parsons/pull/52/files#diff-ee3aa4f1f607ee3259fa9751ca3bf2b3ebdcced0c31d5722a09928744d36ee84R40
* and `auth_name` has never done anything (from the initial commit) https://github.com/move-coop/parsons/blame/fd9a816fdac0e3636cb0290da35322f4ae2925e7/parsons/ngpvan/van.py#L42

Reply:
okay, in that case it seems unlikely we'll break anything by making this change (I'll still warn folks in the release notes though)

Reply:
for sure! if people were passing it because they thought it did something, then this PR will lead to the code throwing an exception!